test/cpp/qps/server_async.cc

```diff
@@ -90,7 +90,7 @@ class AsyncQpsServerTest final : public grpc::testing::Server {
 
     int num_threads = config.async_server_threads();
     if (num_threads <= 0) {  // dynamic sizing
-      num_threads = cores();
+      num_threads = std::min(64, cores());
```

Comment:
I think this is fine for the tests, but I'm worried about the impact on actual benchmarking.

What if we added a 'max_server_threads' command line flag, and plumbed that through from the grpc_cc_test invocation?

Reply:
For benchmarks, the num_threads can be configured through the actual scenario. This cap will only apply if async_server_threads() is not specified in the scenario config (see line 91). Having a flag will probably add confusion between configuring through the scenarios file and configuring through the flag.

Reply:
I think that I can buy that.

Can we add a log line here ('Using 64 cores instead of max 1024 cores') to give ourselves a breadcrumb in case there's a scenario we haven't considered?

And pinging @nanahpang should she have any insights too.

Reply:
Added to existing log in line 94 since I think the log line already contains valuable info.

Reply:
Yes, in the benchmark configuration we always specify the `async_server_threads` number, so it won't affect benchmark results. 

What I am worried is the scalability of this test. Is there a reason why would it cause OOM when thread numbers over 64 and the what is maximum cap it could achieve?

Reply:
The maximum cap is machine-dependent. On Forge, we only get up to 12 GB of memory by default which can handle around 100 threads. I discussed with @ctiller and we decided to make this 64 threads by default since there isn't really anything new being tested by scaling it further.

Reply:
I could add a specification that this is to prevent OOMs on Forge, but that might be too google-specific for the code?